cmd/cloud_sql_proxy/cloud_sql_proxy.go

```diff
@@ -176,6 +176,11 @@ func checkFlags(onGCE bool) error {
 		}
 		return nil
 	}
+
+	if tokF, tok := *tokenFile, *token; tokF != "" || tok != "" {
```

Comment:
Setting the GOOGLE_APPLICATION_CREDENTIALS environment variable is supposed to be equivalent to setting -credential_file. Can you also check if `os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")` is not ""?

Also a style nit: there's no need to store to local variables here. Just compare the values directly:

if *token != "" || *tokenFile != "" || os.Getenv("GOOGLE_APPLICATION_CREDENTIALS") != "" {